### PR TITLE
EXPERIMENTAL: channel upgrade support

### PR DESCRIPTION
**Pull request by rustyrussell**
Based on #4520, though we don't currently use quiescence to force an upgrade.

This is an implementation of https://github.com/lightningnetwork/lightning-rfc/pull/868, though much of it is more about the mechanics of the upgrade we actually end up supporting (option_static_remotekey).

### CODE REVIEW

**Comment by cdecker** [comment]
Sorry for the rebase noise, but a merge caused the SQL files to conflict again.

ACK https://github.com/ElementsProject/lightning/pull/4532/commits/9fb05c03caeea848d1ad5f9d88c07a8e143a593f

File: wallet/db.c [hunk 1]
```diff
@@ -717,6 +717,17 @@ static struct migration dbmigrations[] = {
...
+    /* Channels can now change their type at specific commit indexes. */
+    {SQL("ALTER TABLE channels ADD local_static_remotekey_start BIGINT DEFAULT 0"),
+     NULL},
+    {SQL("ALTER TABLE channels ADD remote_static_remotekey_start BIGINT DEFAULT 0"),
+     NULL},
+    /* Set counter past 2^48 if they don't have option */
+    {SQL("UPDATE channels SET"
+	 " remote_static_remotekey_start = 9223372036854775807,"
+	 " local_static_remotekey_start = 9223372036854775807"
+	 " WHERE option_static_remotekey is 0"),
```
Comment (cdecker) [inline wallet/db.c:729]: This is not portable across databases. Please use an equal here.

**Review by cdecker — CHANGES_REQUESTED** [review]

File: wallet/db.c (hunk 1 above)
Comment (cdecker) [reply wallet/db.c:729]: For referenc, this fails locally on my machine running psql v10.15.

Comment (rustyrussell) [reply wallet/db.c:729]: I hate that it worked on sqlite3.  Fixed.

**Comment by rustyrussell** [comment]
Simple fix for sql non-portability, and rebased on new quiescence branch.

**Comment by rustyrussell** [comment]
Rebase, and remove first patch (will go into separate PR, but only the rest was acked by @cdecker so I'd rather keep it simple)

**Comment by rustyrussell** [comment]
Fixed elements: we were not ignoring fee outputs in "onchaind: don't rely on knowing option_static_remotekey for unknown commitments.".

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/d4809d8aaee739be16b5bad08f54000013e9bd86

(Since fix is trivial, and rest has been reviewed already)

**Comment by rustyrussell** [comment]
Finally fixed all the dumb trivial stuff. Self-ack

Ack https://github.com/ElementsProject/lightning/commit/519adf773c13708698776c53818960d3d1839404

**Pull request merged**